### PR TITLE
Remove unneeded use of `chmod 777`

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Update uses of `chmod 777` to use more specific permissions where appropriate.

## PR Context

Chmod 777 give read, write and execute to everyone.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1 [hunk 1]
```diff
@@ -10,8 +10,8 @@ Describe "Tests Get-Command with relative paths and wildcards" -Tag "CI" {
         #$null = New-Item -ItemType File -Path (Join-Path $TestDrive WildCardCommandA.exe) -ErrorAction Ignore
         #$null = New-Item -ItemType File -Path (Join-Path $TestDRive WildCardCommand[B].exe) -ErrorAction Ignore
         if ( $IsLinux -or $IsMacOS ) {
-            /bin/chmod 777 "$file1"
-            /bin/chmod 777 "$file2"
+            /bin/chmod u+x "$file1"
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1:13]: I'm not sure whether `Setup` runs as the current user, but if it does, this change should be ok. As this is test code, I would suggest that MSRC is being a bit too sensitive. I'm not positive about how Pester removes files, and relying on umask to set the permissions appropriately is not a good idea. I expect the whole purpose of this setting is to ensure that the file can be deleted, execution perms may not be needed at all.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Alias.Tests.ps1 [hunk 2]
```diff
@@ -120,7 +120,7 @@ Describe "Export-Alias DRT Unit Tests" -Tags "CI" {
 		}
 		else
 		{
-			chmod 777 $fulltestpath
+			chmod u+rw $fulltestpath
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Alias.Tests.ps1:123]: i expect this is the same, and execute permission is probably not needed, but `rw` will definitely be needed for cleanup - would `rw-rw-rw-` be more palatable?

Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Alias.Tests.ps1:123]:
```suggestion
			chmod a+rw $fulltestpath
```

File: test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1 (hunk 1 above)
Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1:13]:
```suggestion
            /bin/chmod a+rw "$file1"
```

File: test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1 [hunk 3]
```diff
@@ -10,8 +10,8 @@ Describe "Tests Get-Command with relative paths and wildcards" -Tag "CI" {
         #$null = New-Item -ItemType File -Path (Join-Path $TestDrive WildCardCommandA.exe) -ErrorAction Ignore
         #$null = New-Item -ItemType File -Path (Join-Path $TestDRive WildCardCommand[B].exe) -ErrorAction Ignore
         if ( $IsLinux -or $IsMacOS ) {
-            /bin/chmod 777 "$file1"
-            /bin/chmod 777 "$file2"
+            /bin/chmod a+rw "$file1"
+            /bin/chmod u+x "$file2"
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1:14]:
```suggestion
            /bin/chmod a+rw "$file2"
```

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**